api/src/main/java/org/openmrs/module/fhir2/api/search/param/ObservationSearchParams.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.search.param;
+
+import java.util.HashSet;
+
+import ca.uhn.fhir.model.api.Include;
+import ca.uhn.fhir.rest.api.SortSpec;
+import ca.uhn.fhir.rest.param.*;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@EqualsAndHashCode(callSuper = true)
+public class ObservationSearchParams extends BaseResourceSearchParams {
+	
+	private ReferenceAndListParam encounter;
+	
+	private ReferenceAndListParam patient;
+	
+	private ReferenceAndListParam hasMember;
+	
+	private TokenAndListParam valueConcept;
+	
+	private DateRangeParam valueDate;
+	
+	private QuantityAndListParam valueQuantity;
+	
+	private StringAndListParam valueString;
+	
+	private DateRangeParam date;
+	
+	private TokenAndListParam code;
+	
+	private TokenAndListParam category;
+	
+	/**
+	 * Custom AllArgsConstructor (instead of using @AllArgsConstructor annotation) that supports super
+	 * class instantiation
+	 */
+	public ObservationSearchParams(ReferenceAndListParam encounterReference, ReferenceAndListParam patientReference,
```

Comment:
lombok doesn't support super class field instantiation via the AllArgsConstructor annotation (https://github.com/projectlombok/lombok/issues/1229), so if we want to go this route we'll need to define our own, for each class that inherits from BaseResourceSearchParams